static/app/views/dashboards/widgetBuilder/components/visualize.tsx

```diff
@@ -59,10 +62,65 @@ function Visualize() {
     state.displayType !== DisplayType.BIG_NUMBER;
   const numericSpanTags = useSpanTags('number');
   const stringSpanTags = useSpanTags('string');
-  if (state.dataset === WidgetType.SPANS && isChartWidget) {
-    tags = numericSpanTags;
-  } else if (state.dataset === WidgetType.SPANS && !isChartWidget) {
-    tags = stringSpanTags;
+
+  // Span column options are explicitly defined and bypass all of the
+  // fieldOptions filtering and logic used for showing options for
+  // chart types.
+  let spanColumnOptions;
+  if (state.dataset === WidgetType.SPANS) {
+    // Explicitly merge numeric and string tags to ensure filtering
+    // compatibility for timeseries chart types.
+    tags = {...numericSpanTags, ...stringSpanTags};
+
+    const columns =
+      state.fields
+        ?.filter(field => field.kind === FieldValueKind.FIELD)
+        .map(field => field.field) ?? [];
+    spanColumnOptions = [
+      // Columns that are not in the tag responses, e.g. old tags
+      ...columns
+        .filter(
+          column =>
+            column !== '' &&
+            !stringSpanTags.hasOwnProperty(column) &&
+            !numericSpanTags.hasOwnProperty(column)
+        )
+        .map(column => {
+          return {
+            label: prettifyTagKey(column),
+            value: column,
+            textValue: column,
+            trailingItems: <TypeBadge kind={classifyTagKey(column)} />,
+          };
+        }),
+      ...Object.values(stringSpanTags).map(tag => {
+        return {
+          label: tag.name,
+          value: tag.key,
+          textValue: tag.name,
+          trailingItems: <TypeBadge kind={FieldKind.TAG} />,
```

Comment:
hmm i'm noticing that the trailing badge isn't showing up for spans, was this indended
![image](https://github.com/user-attachments/assets/b908f87a-2458-46c7-a1de-059cc17253f5)

Reply:
Kind of, if you look at explore the visualize aggregates section doesn't put in the trailing items (because these are only number tags) but I've updated this and stuck them in anyways for consistency. If someone has feedback that they shouldn't be there I can remove them